src/HtmlReader/index.tsx

```diff
@@ -123,13 +153,26 @@ export default function useHtmlReader(args: ReaderArguments): ReaderReturn {
         autoGeneratePositions: false,
       },
       api: {
-        getContent: getContent as any, //TODO: fix this casting,
-      } as any, //TODO: fix this casting,,
+        getContent: getContent as GetContent,
+        updateCurrentLocation: (location: Locator) => {
+          // This is needed so that setReadingLocation has the updated "reader" value.
+          dispatch({ type: 'LOCATION_CHANGED', location: location });
```

Comment:
@kristojorg Turns out we do need this to trigger an update to the "reader" object for the setReadingLocation function. This makes things much easier to manage!  Instead of calling for an update on multiple places (button click, TOC navigation, scrolling), this method gets triggered on all user actions on the page.